includes/admin/post-types/meta-boxes/class.llms.meta.box.order.transactions.php

```diff
@@ -198,7 +199,9 @@ private function save_transaction( $post_id ) {
 			return $this->add_error( __( 'Refund Error: Missing or invalid payment amount', 'lifterlms' ) );
 		}
 
-		$order = new LLMS_Order( $post_id );
+		$order        = new LLMS_Order( $post_id );
+		$payment_type = $order->has_trial() && ! $order->get_last_transaction( 'llms-txn-succeeded', 'trial' ) ?
```

Comment:
I think we also need to check that the trial has a payment amount. Otherwise for a free trial, this is applied against the trial vs. single incorrectly.

Reply:
Actually since there is already a transaction added during the initial creation of the Order during checkout, it may be fixing that so it's marked as a `trial` payment type vs. single.